pkg/gits/interface.go

```diff
@@ -54,6 +54,8 @@ type GitProvider interface {
 
 	ListCommitStatus(org string, repo string, sha string) ([]*GitRepoStatus, error)
 
+	ListCommits(owner string, repo string, opt *github.CommitsListOptions) ([]*github.RepositoryCommit, error)
```

Comment:
This should be git provider independent. Please define a generic type which can be reused by all providers instead of `github.CommitListOptions` and respectively `github.RepositoryCommit`.